src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenThatWeWantToGetDependenciesViaDesignTimeBuild.cs

```diff
@@ -12,388 +18,129 @@ public class GivenThatWeWantToGetDependenciesViaDesignTimeBuild
-                            { MetadataKeys.ParentTarget, "net45" }
-                        })
-                }
-            };
-
-            Assert.True(task.Execute());
-
-            var item = Assert.Single(task.PackageDependenciesDesignTime);
-
-            Assert.Equal("mockPackageUnresolved/1.0.0", item.ItemSpec);
-            Assert.Equal("mockPackageUnresolved", item.GetMetadata(MetadataKeys.Name));
-            Assert.Equal("1.0.0", item.GetMetadata(MetadataKeys.Version));
-            Assert.Equal("some path", item.GetMetadata(MetadataKeys.Path));
-            Assert.Equal("", item.GetMetadata(MetadataKeys.ResolvedPath));
-            Assert.Equal("Warning", item.GetMetadata(MetadataKeys.DiagnosticLevel));
-            Assert.False(item.GetBooleanMetadata(MetadataKeys.IsImplicitlyDefined));
-            Assert.False(item.GetBooleanMetadata(PreprocessPackageDependenciesDesignTime.ResolvedMetadata));
-        }
-
-        [Fact]
-        public void ItShouldIdentifyDefaultImplicitPackages()
-        {
-            var task = new PreprocessPackageDependenciesDesignTime
-            {
-                TargetFramework = "net45",
-                DefaultImplicitPackages = "DefaultImplicit",
-                PackageDefinitions = new ITaskItem[]
-                {
-                    new MockTaskItem(
-                        itemSpec: "DefaultImplicit/1.0.0",
-                        metadata: new Dictionary<string, string>
-                        {
-                            { MetadataKeys.Name, "DefaultImplicit" },
-                            { MetadataKeys.Version, "1.0.0" },
-                            { MetadataKeys.Path, "some path" },
-                            { MetadataKeys.ResolvedPath, "" },
-                            { MetadataKeys.Type, "Package" }
-                        })
-                },
-                PackageDependencies = new ITaskItem[]
-                {
-                    new MockTaskItem(
-                        itemSpec: "DefaultImplicit/1.0.0",
-                        metadata: new Dictionary<string, string>
-                        {
-                            { MetadataKeys.ParentTarget, "net45" }
-                        })
-                }
-            };
-
-            Assert.True(task.Execute());
-
-            var item = Assert.Single(task.PackageDependenciesDesignTime);
-
-            Assert.Equal("DefaultImplicit/1.0.0", item.ItemSpec);
-            Assert.True(item.GetBooleanMetadata(MetadataKeys.IsImplicitlyDefined));
+            string projectAssetsJsonPath = Path.GetTempFileName();
+            string projectCacheAssetsJsonPath = Path.GetTempFileName();
+            var assetsContent = @"{
+  `version` : 3,
+  `targets` : {
+    `net6.0` : {
+      `Newtonsoft.Json/12.0.1` : {
+        `type` : `package`,
+        `compile` : {
+            `lib/netstandard2.0/Newtonsoft.Json.dll` : {
+                `related` : `.pdb;.xml`
+            }
+        },
```

Comment:
@dsplaisted what values can be passed to the json file to avoid the null value in `packagePath `from `_packagePathResolver`

https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/NuGetPackageResolver.cs#L49